src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -106,188 +112,209 @@ void EnCow_Init(Actor* thisx, PlayState* play) {
         this->actor.flags &= ~ACTOR_FLAG_16;
         Message_CloseTextbox(play);
-        this->actionFunc = func_809DF96C;
+        this->actionFunc = EnCow_Idle;
     }
 }
 
-void func_809DF730(EnCow* this, PlayState* play) {
+void EnCow_GiveMilkEnd(EnCow* this, PlayState* play) {
     if (Actor_TextboxIsClosing(&this->actor, play)) {
         this->actor.flags &= ~ACTOR_FLAG_16;
-        this->actionFunc = func_809DF96C;
+        this->actionFunc = EnCow_Idle;
     }
 }
 
-void func_809DF778(EnCow* this, PlayState* play) {
+void EnCow_GiveMilkWait(EnCow* this, PlayState* play) {
     if (Actor_HasParent(&this->actor, play)) {
         this->actor.parent = NULL;
-        this->actionFunc = func_809DF730;
+        this->actionFunc = EnCow_GiveMilkEnd;
     } else {
         Actor_OfferGetItem(&this->actor, play, GI_MILK, 10000.0f, 100.0f);
     }
 }
 
-void func_809DF7D8(EnCow* this, PlayState* play) {
+void EnCow_GiveMilk(EnCow* this, PlayState* play) {
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play)) {
         this->actor.flags &= ~ACTOR_FLAG_16;
         Message_CloseTextbox(play);
-        this->actionFunc = func_809DF778;
+        this->actionFunc = EnCow_GiveMilkWait;
         Actor_OfferGetItem(&this->actor, play, GI_MILK, 10000.0f, 100.0f);
     }
 }
 
-void func_809DF870(EnCow* this, PlayState* play) {
+void EnCow_CheckForEmptyBottle(EnCow* this, PlayState* play) {
     if ((Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT) && Message_ShouldAdvance(play)) {
         if (Inventory_HasEmptyBottle()) {
             Message_ContinueTextbox(play, 0x2007);
-            this->actionFunc = func_809DF7D8;
+            this->actionFunc = EnCow_GiveMilk;
         } else {
             Message_ContinueTextbox(play, 0x2013);
-            this->actionFunc = func_809DF6BC;
+            this->actionFunc = EnCow_TalkEnd;
         }
     }
 }
 
-void func_809DF8FC(EnCow* this, PlayState* play) {
+void EnCow_Talk(EnCow* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, play)) {
-        this->actionFunc = func_809DF870;
+        this->actionFunc = EnCow_CheckForEmptyBottle;
     } else {
         this->actor.flags |= ACTOR_FLAG_16;
         func_8002F2CC(&this->actor, play, 170.0f);
         this->actor.textId = 0x2006;
     }
-    func_809DF494(this, play);
+
+    EnCow_UpdateAnimation(this, play);
 }
 
-void func_809DF96C(EnCow* this, PlayState* play) {
+void EnCow_Idle(EnCow* this, PlayState* play) {
     if ((play->msgCtx.ocarinaMode == OCARINA_MODE_00) || (play->msgCtx.ocarinaMode == OCARINA_MODE_04)) {
-        if (DREG(53) != 0) {
-            if (this->unk_276 & 4) {
-                this->unk_276 &= ~0x4;
-                DREG(53) = 0;
+        if (R_EPONA_CALLED) {
+            if (this->cowFlags & COW_FLAG_DONT_GIVE_MILK) {
+                this->cowFlags &= ~COW_FLAG_DONT_GIVE_MILK;
+                R_EPONA_CALLED = false;
```

Comment:
I think the comment there was nice to have because imo it's not trivial what this achieves (working properly with other things reacting to epona song)

Also not convinced by naming the reg after calling epona when it's really "has played epona song" isn't it?

Reply:
Iirc this also handles having a single cow give milk if there are several in range, but it's been a long time I kinda forgot

And about the reg I mean, cows react to playing the song, and the code listens on that reg does it not? (Again a long time has passed)

Reply:
ah, yeah I did miss that the cow is listening for that reg to continue its own process. Thought it was just unsetting it instantly to prevent epona from spawning.

Ill change the name back, but still not sure on the comment, maybe it can be reworded. It still seems to me like its always the actor's responsibility to set the flag to false, since the message code (which sets it to true) doesnt handle that.

What is the "next tick" in the original comment referring to? It just gets unset immediately if a cow is giving milk, not sure what is happening on the next tick/frame

Reply:
For the tick thing I don't remember exactly but I do think there is a one frame delay before actions are taken, maybe the previous discussion on this will help (I don't remember well enough to write an explanation right now) https://github.com/zeldaret/oot/pull/1492#discussion_r1070426005

Reply:
alright i understand now and have pushed a change.

I changed `COW_WONT_GIVE_MILK` -> `COW_FLAG_FAILED_TO_GIVE_MILK`. I think this better expresses the intent of why the flag exists/how it behaves in this interaction.
I also wrote a pretty lengthy comment that expands alot on the original. I think it is warranted for how complex the interaction is (especially cause I could not even notice the complexity of the interaction with the old comment)